### DISCUSSION TITLE
Multilib Library path.

### DISCUSSION THREAD

**Discussion by CRKatri** [ProcursusTeam/Procursus#829] [Ideas]
In order to improve cross compilation ability we could setup multilib library paths. Because MacOS and iOS share triples though, we may want to use the `MEMO_TARGET`. To make this work well, we would set libdir to `$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/$(MEMO_TARGET)`, and add that path to the end of `DYLD_FALLBACK_LIBRARY_PATH` ([dyld(1)](https://man.cameronkatri.com/MacOSX12b1/dyld.1)), so that if the library isn't found in the normal path it will check the multilib location. This is similar to how Debian does it. One issue with this is that `MacOS` targets install to `/opt/procursus` so it may be impossible for iOS libs installed on MacOS, but for installing arm64e and arm64 on MacOS it would work well, and AppleTV/WatchOS on iOS and vice versa.

**Comment by Diatrus** [comment]
This is a great idea I wanted to do for a long time, but not super feasible when it comes to the workload required for this.

**Reply by CRKatri** [reply]
It shouldn't be that big a work load.
1. Add `DYLD_FALLBACK_PATH` to `profile.d`.
2. Set libdir to the new path in `DEFAULT_*_FLAGS`
3. Libraries will use the new path as packages are updated.

**Comment by CRKatri** [comment]
@Diatrus I just push `ck/multilib-experiment`, which has some starter stuff, but it still needs a lot of work, it didn't occur to me that the `*-package` targets hard code `$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libwhatever` when packaging. It may be better to do this during the rewrite that I've been hearing about.

**Reply by CRKatri** [reply]
FYI: `DYLD_FALLBACK_LIBRARY_PATH` is ignored for `setuid` binaries on MacOS (possibly iOS too, but the section to prune them is in an if block for `TARGET_OS_OSX` (src/dyld2.cpp)) so any package with `setuid` bins may need some special depends on libraries that are multilib'd.
